ledger/src/blockstore.rs

```diff
@@ -170,6 +174,50 @@ impl<T> AsRef<T> for WorkingEntry<T> {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub struct LastFECSetCheckResults {
+    block_id: Option<Hash>,
+    is_retransmitter_signed: bool,
+}
```

Comment:
Can `block_id` be `Some(...)` and `is_retransmitter_signed` also be `false`?
if not, then probably this should be an `enum` instead of `struct`.

Reply:
Yes, if chained merkle roots are not rolled out for this slot, but padding to 32+ data shreds is active, then `block_id` will be `Some(...)` but `is_retransmitter_signed` will be false.